daemon/containerd/image_push.go

```diff
@@ -62,13 +63,20 @@ func (i *ImageService) PushImage(ctx context.Context, targetRef reference.Named,
 	store := i.client.ContentStore()
 
 	resolver, tracker := i.newResolverFromAuthConfig(ctx, authConfig)
-	progress := pushProgress{Tracker: tracker}
+	pp := pushProgress{Tracker: tracker}
 	jobsQueue := newJobs()
 	finishProgress := jobsQueue.showProgress(ctx, out, combinedProgress([]progressUpdater{
-		&progress,
+		&pp,
 		pullProgress{ShowExists: false, Store: store},
 	}))
-	defer finishProgress()
+	defer func() {
+		finishProgress()
+		if tagged, ok := targetRef.(reference.Tagged); ok {
+			if retErr == nil {
```

Comment:
nit: retErr check could be first to avoid the type assertion on error. (although Go compiler is probably smart enough to optimize this so feel free to ignore)

Reply:
Thanks, did the switcheroo